aws-redshiftserverless-workgroup/src/main/java/software/amazon/redshiftserverless/workgroup/DeleteHandler.java

```diff
@@ -31,43 +36,94 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
                         proxy.initiate("AWS-RedshiftServerless-Workgroup::Delete", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
                                 .translateToServiceRequest(Translator::translateToDeleteRequest)
                                 .backoffDelay(BACKOFF_STRATEGY)
-                                .makeServiceCall(this::deleteWorkgroup)
-                                .stabilize(this::isWorkgroupStable)
+                                .makeServiceCall((awsRequest, sdkProxyClient) -> {
+                                    DeleteWorkgroupResponse awsResponse = this.deleteWorkgroup(awsRequest, sdkProxyClient);
+
+                                    logger.log(String.format("%s : %s has successfully been deleted.",
+                                            ResourceModel.TYPE_NAME, awsResponse.workgroup().workgroupName()));
+                                    logger.log(awsResponse.toString());
+
+                                    return awsResponse;
+                                })
+                                .stabilize(this::isWorkgroupDeleted)
                                 .handleError(this::deleteWorkgroupErrorHandler)
-                                .progress()
+                                .done(awsResponse -> {
+                                    return ProgressEvent.progress(Translator.translateFromDeleteResponse(awsResponse), callbackContext);
+                                })
                 )
+                .then(progress ->
+                        proxy.initiate("AWS-RedshiftServerless-Workgroup::ReadNameSpace", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                                .translateToServiceRequest(Translator::translateToReadNamespaceRequest)
+                                .backoffDelay(BACKOFF_STRATEGY)
+                                .makeServiceCall((awsRequest, sdkProxyClient) -> {
+                                    GetNamespaceResponse awsResponse = this.readNamespace(awsRequest, sdkProxyClient);
+
+                                    logger.log(String.format("%s : %s has successfully been created.",
+                                            ResourceModel.TYPE_NAME, awsResponse.namespace().namespaceName()));
+                                    logger.log(awsResponse.toString());
 
+                                    return awsResponse;
+                                })
+                                .stabilize(this::isNamespaceStable)
```

Comment:
We need to check if the workgroup is stable or not before deleting the workgroup. Namespace stability check is not required